lib/optimizely/decision_service.rb

```diff
@@ -156,65 +145,43 @@ def get_variation_for_feature_experiment(feature_flag, user_id, attributes = nil
       #
       # Returns a hash with the experiment and variation where visitor will be bucketed
       # or nil if the user is not bucketed into any of the experiments on the feature
-
       feature_flag_key = feature_flag['key']
-      unless feature_flag['experimentIds'].empty?
-        # check if experiment is part of mutex group
-        experiment_id = feature_flag['experimentIds'][0]
-        experiment = @config.experiment_id_map[experiment_id]
-        unless experiment
-          @config.logger.log(
-            Logger::DEBUG,
-            "Feature flag experiment with ID '#{experiment_id}' is not in the datafile."
-          )
-          return nil
-        end
-
-        group_id = experiment['groupId']
-        # if experiment is part of mutex group we first determine which experiment (if any) in the group the user is part of
-        if group_id and @config.group_key_map.has_key?(group_id)
-          group = @config.group_key_map[group_id]
-          bucketed_experiment_id = @bucketer.find_bucket(user_id, group_id, group['trafficAllocation'])
-          if bucketed_experiment_id.nil?
+      if feature_flag['experimentIds'].empty?
+        @config.logger.log(
+          Logger::DEBUG,
+          "The feature flag '#{feature_flag_key}' is not used in any experiments."
+        )
+        return nil
+      else
```

Comment:
nitpick: because you return out of the previous `if` clause, you can skip the `else` so we reduce the nesting